lib/utils/templatize.js

```diff
@@ -364,7 +364,7 @@ function createTemplatizerClass(template, templateInfo, options) {
  */
 function addPropagateEffects(template, templateInfo, options) {
   let userForwardHostProp = options.forwardHostProp;
-  if (userForwardHostProp) {
+  if (userForwardHostProp && templateInfo.hasHostProps) {
```

Comment:
Let's add a comment here or perhaps just in the doc for the function explaining what this does so it's clear why it's only related to `hostProps`.